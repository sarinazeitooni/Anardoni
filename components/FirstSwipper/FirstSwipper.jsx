import React from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import Card from "../card/Card";
import Link from 'next/link';
import cardData from "../../data/CardData";
import style from './scss/first-swipper.module.scss';
const FirstSwipper = () => {
    let data = cardData;
    return (
        <div className={style['swipper-container']}>
                <Swiper spaceBetween={1} slidesPerView={3.1}>
                    {data.map((card) =>
                        <SwiperSlide>
                        <Link href={{
                            pathname : '/cards/[id]',
                            query : {id : card.id}
                        }}>
                            <a>
                            <Card id={card.id} title={card.title} subTitle={card.subTitle}
                                  detailImage={card.detailImage} containerImage={card.containerImage}/>
                            </a>
                        </Link>
                        </SwiperSlide>
                    )}
                </Swiper>
        </div>
    )
};
export default FirstSwipper;