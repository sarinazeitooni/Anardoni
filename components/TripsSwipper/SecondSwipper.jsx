import React, {useEffect, useState} from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import style from './scss/trips-swipper.module.scss';
import messages from "./messages/messages";
import {useSelector, useDispatch} from "react-redux";
import SwiperSkeleton from "../skeleton/SwiperSkeleton";
import SmallCardContainer from "../smallCard/SmallCardContainer";
import swiperAction from "../../redux/action/Swiperaction";

function SecondSwipper({title}) {
    const [swipperData, setSwipperData] = useState(null);
    const dispatch = useDispatch();
    const swipperDataForTimeOut = useSelector(state => state.swipper);
    useEffect(() => {
        dispatch(swiperAction());
        setTimeout(() => {
            setSwipperData(swipperDataForTimeOut);

        }, 1000);
    }, [swipperDataForTimeOut]);
    return (
        <div className={style['second-swipper-container']}>
            <div className={style['swipper-title']}>
                <div className={style['right-title']}>{title}</div>
                <div className={style['left-title']}><span>{messages.more}</span></div>
            </div>
            <Swiper spaceBetween={0} slidesPerView={7}>
                {!swipperData && (
                    <SwiperSkeleton slides={10}/>
                )}
                {swipperData && (swipperData.map((item) => {
                    return (
                        <SwiperSlide key={item.id}>
                            <SmallCardContainer title={item.title} detail={item.detail} cardImage={item.url}/>
                        </SwiperSlide>
                    )
                }))
                }
            </Swiper>
        </div>)
}
export default SecondSwipper;