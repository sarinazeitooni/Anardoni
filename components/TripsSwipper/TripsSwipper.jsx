import React, {useEffect, useState} from 'react';
import {Swiper, SwiperSlide} from 'swiper/react';
import style from './scss/TripsSwipper.module.scss';
import messages from "./messages/messages";
import { useSelector , useDispatch } from "react-redux";
import MySkeleton from "../skeleton/Skeleton";
import SmallCardContainer from "../smallCard/SmallCardContainer";
import swiperAction from "../../redux/action/action";
function SecondSwipper({title}){
    const [swipperData , setSwipperData] = useState(null);
    const dispatch = useDispatch();
    const swipperDataForTimeOut = useSelector(state => state.swipper);
    useEffect(()=>{
        dispatch(swiperAction());
        setTimeout(()=>{
            setSwipperData(swipperDataForTimeOut);

        },1000);
    },[swipperDataForTimeOut])
    return (
        <div className={style['second-swipper-container']}>
            <div className={style['swipper-title']}>
                <div className={style['right-title']}>{title}</div>
                <div className={style['left-title']}><span>{messages.more}</span></div>
            </div>
                <Swiper spaceBetween={0} slidesPerView={7}>
                    {!swipperData && (
                        <MySkeleton slides={10}/>
                    )}
                    {swipperData && (swipperData.map((item)=>{
                        return(
                            <SwiperSlide>
                                <SmallCardContainer title={item.title} detail={item.detail} cardImage={item.url}/>
                            </SwiperSlide>
                        )
                    }))
                    }
                </Swiper>
        </div>)
};
export default SecondSwipper;