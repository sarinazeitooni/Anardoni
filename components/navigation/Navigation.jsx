import React, {useEffect, useState} from 'react';
import HomeTwoToneIcon from '@material-ui/icons/HomeTwoTone';
import SearchTwoToneIcon from '@material-ui/icons/SearchTwoTone';
import LayersIcon from '@material-ui/icons/Layers';
import WhatshotIcon from '@material-ui/icons/Whatshot';
import CategoryTwoToneIcon from '@material-ui/icons/CategoryTwoTone';
import AddCircleIcon from '@material-ui/icons/AddCircle';
import VpnKeyIcon from '@material-ui/icons/VpnKey';
import messages from "./messages/messages";
import style from './scss/navigation.module.scss';
import {useRouter} from 'next/router';
import {useSelector, useDispatch} from "react-redux";
import swiperAction from "../../redux/action/Swiperaction";
import navigationAction from "../../redux/action/navigationAction";
import useDeviceDetect from "../../customHook/useDeviceDetect";
function Navigation() {
    const isMobile = useDeviceDetect;
    const dispatch = useDispatch();
    const router = useRouter();
    const [navigationBtn, setNavigationBtn] = useState(style.items);
    const [NavItemClass, SetNavItemClass] = useState('');
    const navigationBtnTimeout = useSelector(state => state.navigationBtn);
    console.log(isMobile);
    const btnChange =(btn) => {
        dispatch(navigationAction(btn));
    }
    useEffect(() => {
        setTimeout(() => {
            setNavigationBtn(navigationBtnTimeout);
        }, 100);
    }, [navigationBtnTimeout])
    useEffect(() => {
        (router.pathname == '/') ? SetNavItemClass(style.selected) : SetNavItemClass('');
    }, []);

    return (

        <div className={style['navigation-container']}>
            <ul className={style['item-container']}>
                <button onClick={() => {
                    btnChange('first')
                }} className={navigationBtn == 'first' ? style.selected : style.items}><a
                    href='#home'>{messages.home}</a>
                    <HomeTwoToneIcon/></button>
                <button onClick={() => {
                    btnChange('second')
                }} className={navigationBtn == 'second' ? style.selected : style.items}><a
                    href='#apps'>{messages.apps}</a><LayersIcon/>
                </button>
                <button onClick={() => {
                    btnChange('third')
                }} className={navigationBtn == 'third' ? style.selected : style.items}><a
                    href='#games'>{messages.games}</a><WhatshotIcon/>
                </button>
                <button onClick={() => {
                    btnChange('forth')
                }}  className={navigationBtn == 'forth' ? style.selected : style.items}><a
                    href='#categories'>{messages.categories}</a>
                    <CategoryTwoToneIcon/></button>
                <button onClick={() => {
                    btnChange('fifth')
                }} className={navigationBtn == 'fifth' ? style.selected : style.items}><a
                    href='#search'>{messages.search}</a>
                    <SearchTwoToneIcon/></button>
            </ul>
            <div className={style['log-in-container']}>
                <button className={style['right-btn']}>{messages.signup}<AddCircleIcon/></button>
                <span className={style['black-circle']}></span>
                <button className={style['left-btn']}><VpnKeyIcon/>{messages.login}</button>
            </div>
        </div>
    )
}

export default Navigation;