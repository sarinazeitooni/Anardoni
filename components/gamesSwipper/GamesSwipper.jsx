import React from 'react';
import SecondSwipper from "../TripsSwipper/TripsSwipper";
import messages from "./messages/messages";

function GamesSwipper() {
    return (
        <div id='games'>
            <SecondSwipper title={messages.games}/>
        </div>
    )
};
export default GamesSwipper;