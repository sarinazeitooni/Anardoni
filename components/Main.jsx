import React from 'react';
import Navigation from "./navigation/Navigation";
import FirstSwipper from "./FirstSwipper/FirstSwipper";
import TripsSwipper from "./TripsSwipper/TripsSwipper";
import FeedBack from "./feedback/feedback";
import { Provider } from "react-redux";
import {BrowserView , isMobile} from 'react-device-detect';
import messages from "./TripsSwipper/messages/messages";
import MainMobile from "./MainMobile";
import store from '../redux/store/store';
import GamesSwipper from "./gamesSwipper/GamesSwipper";
function Main(){
    return(
     <div>
         <Provider store={store}>
             <BrowserView>
                 <Navigation/>
                 <FirstSwipper/>
                 <FeedBack/>
                 <TripsSwipper title={messages.trips}/>
                 <GamesSwipper/>
             </BrowserView>
             {isMobile && <MainMobile/>}
         </Provider>
     </div>
    )
};
export default Main;