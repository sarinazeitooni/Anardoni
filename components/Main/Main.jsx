import React, {useEffect} from 'react';
import Navigation from "../navigation/Navigation";
import FirstSwipper from "../FirstSwipper/FirstSwipper";
import TripsSwipper from "../TripsSwipper/TripsSwipper";
import FeedBack from "../feedback/feedback";
import messages from "../TripsSwipper/messages/messages";
import GamesSwipper from "../gamesSwipper/GamesSwipper";
function Main() {
    return (
        <React.Fragment>
            <Navigation/>
            <FirstSwipper/>
            <FeedBack/>
            <TripsSwipper title={messages.trips}/>
            <FirstSwipper/>
            <GamesSwipper/>
        </React.Fragment>
    )
};
export default Main;