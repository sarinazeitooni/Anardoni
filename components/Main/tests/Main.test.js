import React from 'react';
import { shallow } from 'enzyme';
import Main from "../Main";
import Navigation from "../../navigation/Navigation";
import FirstSwipper from "../../FirstSwipper/FirstSwipper";
import FeedBack from "../../feedback/feedback";
import TripsSwipper from "../../TripsSwipper/TripsSwipper";
import GamesSwipper from "../../gamesSwipper/GamesSwipper";

describe("Main component", () => {
    it("should have 6 child component", () => {
        const main = shallow(<Main/>);
        expect(main.find(<Navigation/>)).toEqual({});
        expect(main.find(<FirstSwipper/>)).toEqual({});
        expect(main.find(<FeedBack/>)).toEqual({});
        expect(main.find(<TripsSwipper/>)).toEqual({});
        expect(main.find(<FirstSwipper/>)).toEqual({});
        expect(main.find(<GamesSwipper/>)).toEqual({});
        expect(main).toMatchSnapshot();
    });
});