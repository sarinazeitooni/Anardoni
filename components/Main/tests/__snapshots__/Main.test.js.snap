// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should have 6 child component 1`] = `ShallowWrapper {}`;
